Add guidelines for running CMIP7 experiments (#842)
ACCESS-NRI/access-esm1.6-configs: documentation/docs/pages/configs_experiments/fast_track_guidelines.md

On completion, ESM1.6 Payu experiments for the CMIP7 Assessment Fast Track will need to be uploaded to the [access-esm1.6-dev-experiments](https://github.com/ACCESS-Community-Hub/access-esm1.6-dev-experiments) GitHub repository.

Payu *configurations* and *experiments* both contain all the information required to run a model, and either can be used to start a simulation.

--- 

# Instructions and guidelines
## Cloning and running an experiment
In most cases, ACCESS-NRI and CSIRO will provide you with a branch on the [access-esm1.6-dev-experiments](https://github.com/ACCESS-Community-Hub/access-esm1.6-dev-experiments) GitHub repository which you can clone, run, and push the runlogs back to. 

For example, to clone the *example-experiment* [branch](https://github.com/ACCESS-Community-Hub/access-esm1.6-dev-experiments/tree/example-experiment), you would first load the payu module, and use:
```

> payu clone git@github.com:ACCESS-Community-Hub/access-esm1.6-dev-experiments.git -B example-experiment <control directory name>

Reviewer: Given the problem people seem to have with the separation between control directory names and branches I think it makes sense to suggest a control directory name. If we don't just use the default repo name maybe something like `esm1.6-cmip7-fast-track`.

Reviewer: Aside: is `access-esm1.6-dev-experiments` a good name? @ccarouge should we rename it to `access-esm1.6-experiments` or create a new repo for this express purpose?

Author: I'm unsure about specifying a single default as people are planning to run several different fast track experiments at once. If we do suggest a directory name here, would naming it after the specific experiment be easiest?

Reviewer: If the plan is to push the experiments back to a central repository the choice of control directory name should be chosen for clarity of the experiment archive name.

Author: Just to make sure I properly understand, do you mean it should be named based on the central repository name?

Reviewer: It depends on the central repo name. If it's a good descriptive git, then sure. Above I suggested `esm1.6-cmip7-fast-track`, but equally it could be `cmip7-esm1.6-contrib` or similar, to highlight it is a community contribution.

I don't really have a strong opinion, just pointing out that there is the opportunity here to enforce/suggest something systematic that could make it easier for users later to identify this class of experiment, given the branch name is going to be the bit that separates one class of experiment from another.

Or maybe it isn't necessary? In which case it should be standardised. @ccarouge may have an opinion.

Reviewer 2: A few things:

- Repo name: Yes, we should rename the experiment's repo name. GitHub handles redirection so it won't break current workflows. Now to decide whether it makes sense to split between FastTrack and other CMIP7 experiments? So it could be `access-esm1.6-experiments-cmip7` or `access-esm1.6-experiments-cmip7-fasttrack` (this one starts to be quite long!).
- Control directory name: @blimlim Aidan came to my office this morning and gave me a full explanation. The full name of the experiment in intake catalogs for example etc is: `<control_dir>_<branch-name>_uuid`. So @aidanheerdegen would like the <control_dir> part to be standardised (for new experiments). I'm not sure what yet. But I have a question for @aidanheerdegen : what do people do when they run several experiments at the same time? We recommend people to put their payu control directories under a \<model> directory. But then, if we ask them to have the same control directory name for all their cmip7 experiments (using branches to switch between them), they can't run several at the same time. Or am I missing something?

```

!!! warning
    If you are cloning an experiment which has been set up for you on the [access-esm1.6-dev-experiments](https://github.com/ACCESS-Community-Hub/access-esm1.6-dev-experiments) GitHub repository, it's important that you don't change the local branch name using the `-b <local-branch-name>` option from the `payu clone` command.*

> Once the experiment has been cloned, you can navigate to the control directroy and run it using the usual `payu run -n <n runs>` command.

Reviewer: ```suggestion
Once the experiment has been cloned, you can `cd` into the control directory cloned in the command above and run it using the usual `payu run -n <n runs>` command.
```

## Output syncing
Model outputs and restarts should be synced to a location on `/g/data` to prevent loss of data during the automatic cleanup of files on `/scratch`. We recommend configuring Payu to automatically sync the outputs and restarts to a location on `/g/data` by enabling the `sync` option in the `config.yaml`:

```diff
# Sync options for automatically copying data from ephemeral scratch space to
# longer term storage
sync:
-   enable: False # set path below and change to true
+   enable: True # set path below and change to true
    restarts: True
    base_path: <Location on /g/data>
```

With the above changes, Payu will create a new directory matching your experiment's name under the location specified by `base_path`. Outputs and restarts will then be copied to this location at the end of each run segment. For more information on automatic syncing, please see the [Payu documentation](https://payu.readthedocs.io/en/stable/config.html#:~:text=the%20PBS%20script.-,sync,-Sync%20archive%20to).

> ACCESS-NRI and CSIRO staff will work with you to determine the best location for syncing your data.

Reviewer: Do we want to suggest using working group projects for this? Or perhaps a temporary shared space for this purpose @ccarouge?

Reviewer 2: As a general rule, I would prefer people to use their own storage for this. Then, when it's publication time, they can work with our data team and NCI's for the cmorisation, giving access to people etc. 

CSIRO will not be involved in the storage or publication of community contributions to FastTrack or CMIP7 in general.

Author: Thanks for clarifying @ccarouge.  Just want to check if paragraph further below needs to be changed:

> Output and restart files for ESM1.6 fast track experiments are being archived to the `p73` project. Write access to `p73` is restricted and so CSIRO team members will be in contact with you to organise a data copy upon your simulations' completion.

We might need to discuss this in the stand up next week, I'm not sure that everyone's on the same page about storage locations, see: https://forum.access-hive.org.au/t/experiment-proposal-fast-track-abrupt-co-experiments/6321#:~:text=Total%20storage%20required

Author: If I followed yesterday's discussion properly, it sounds like people will use their own storage during the run and copy to p73 at the end, at least for the time being. @ccarouge does that match your understanding?

Reviewer 2: Yes, that's my understanding.

## Output archiving
Output and restart files for ESM1.6 fast track experiments are being archived to the `p73` project. Write access to `p73` is restricted and so CSIRO team members will be in contact with you to organise a data copy upon your simulations' completion.

## Pushing completed experiments back to the repository
Once your simulations are complete, you'll need to push the runlogs back up to the [access-esm1.6-dev-experiments](https://github.com/ACCESS-Community-Hub/access-esm1.6-dev-experiments) GitHub repository using the following steps:

1.  First find the local branch name for each experiment by running the following command in the control directory:
    ```bash
    $ payu branch
    * Current Branch: <experiment-branch-name>
        experiment_uuid: 3e907528-b735-af72-b4v6-51c9b3l27960
    ```
2. Push the runlogs up to the repository:
   ```

>    $ git push origin <experiment-branch-name>

Reviewer: You can just `git push origin HEAD` if already on the correct branch.

Do we need to be worried about people pushing to the wrong branch and deleting work? I suppose not if we're backing up the repo (just checked, and yes seem to be up to date).

Wondering if we need a warning **not** to force push: if there is a problem contact us to work through issues.

Author: I've swapped it over to `git push origin HEAD`. We have a branch protection rule on the experiments repo preventing force pushes, and so I think we should be protected against accidental overwrites
